src/aggregations/pipeline-aggregations/moving-function-aggregation.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+const isNil = require('lodash.isnil');
+
+const PipelineAggregationBase = require('./pipeline-aggregation-base');
+
+const ES_REF_URL =
+    'https://www.elastic.co/guide/en/elasticsearch/reference/master/search-aggregations-pipeline-movfn-aggregation.html';
+
+/**
+ * Given an ordered series of data, the Moving Function aggregation
+ * will slide a window across the data and allow the user to specify
+ * a custom script that is executed on each window of data.
+ * For convenience, a number of common functions are predefined such as min/max, moving averages, etc.
+ *
+ * `moving_fn` aggregations must be embedded inside of a histogram or
+ * date_histogram aggregation.
+ *
+ * [Elasticsearch reference](https://www.elastic.co/guide/en/elasticsearch/reference/master/search-aggregations-pipeline-movfn-aggregation.html)
+ *
+ * @example
+ * const agg = esb.movingFunctionAggregation('the_movfn', 'the_sum')
+ *     .model('holt')
+ *     .window(5)
+ *     .gapPolicy('insert_zeros')
+ *     .settings({ alpha: 0.8 });
+ *
+ * @example
+ * const reqBody = esb.requestBodySearch()
+ *     .agg(
+ *         esb.dateHistogramAggregation('my_date_histo', 'timestamp')
+ *             .interval('day')
+ *             .agg(esb.sumAggregation('the_sum', 'lemmings'))
+ *             // Relative path to sibling metric `the_sum`
+ *             .agg(esb.movingFunctionAggregation('the_movfn', 'the_sum'))
+ *     )
+ *     .size(0);
+ *
+ * @example
+ * const reqBody = esb.requestBodySearch()
+ *     .agg(
+ *         esb.dateHistogramAggregation('my_date_histo', 'timestamp')
+ *             .interval('day')
+ *             // Use the document count as it's input
+ *             .agg(esb.movingFunctionAggregation('the_movfn', '_count'))
+ *     )
+ *     .size(0);
+ *
+ * @param {string} name The name which will be used to refer to this aggregation.
+ * @param {string=} bucketsPath The relative path of metric to aggregate over.
+ * @param {string=} window 	The size of window to "slide" across the histogram.
+ * @param {string=} script The script that should be executed on each window of data.
+ *
+ * @extends PipelineAggregationBase
+ */
+class MovingFunctionAggregation extends PipelineAggregationBase {
+    // eslint-disable-next-line require-jsdoc
+    constructor(name, bucketsPath, window, script) {
+        super(name, 'moving_fn', ES_REF_URL, bucketsPath);
+
+        if (!isNil(window)) this._aggsDef.window = window;
+        if (!isNil(script)) this._aggsDef.script = script;
+    }
+
+    /**
+     * @override
+     * @throws {Error} This method cannot be called on MovingFunctionAggregation
+     */
+    format() {
```

Comment:
I think `format` is supported for this aggregation based on https://github.com/elastic/elasticsearch/blob/v7.10.1/server/src/main/java/org/elasticsearch/search/aggregations/pipeline/MovFnPipelineAggregationBuilder.java#L67.

Reply:
I can't find any mention of this in the documentation. Would you still add it?

Reply:
Elasticsearch documentation is not exhaustive. There are other instances as well where I have had to refer the source code to determine the supported properties.

Reply:
OK, I added the support of format :+1: